service/src/main/java/org/jboss/aerogear/unifiedpush/service/metrics/PushMessageMetricsService.java

```diff
@@ -64,7 +68,11 @@ public void updatePushMessageInformation(FlatPushMessageInformation pushMessageI
     public void appendError(final FlatPushMessageInformation pushMessageInformation, final Variant variant, final String errorMessage) {
         final VariantErrorStatus ves = new VariantErrorStatus(pushMessageInformation, variant, errorMessage);
         pushMessageInformation.getErrors().add(ves);
-        flatPushMessageInformationDao.update(pushMessageInformation);
+        try {
+            flatPushMessageInformationDao.update(pushMessageInformation);
+        } catch (Exception e) {
+            logger.info("A VariantErrorStatus has been already added, for the given Variant in this push job");
```

Comment:
We catch a generic exception here, so the reason for the failure could be different than duplicate key.
I would change this error message and I would add the exception as well.
Something like: 
```java
   logger.info("Failed to save pushMessageInformation: {}", e.getMessage());
   logger.debug("Details:", e);
```